packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/Utilities/StringHelpers.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.Generator.CSharp.Input
+{
+    public static class StringHelpers
+    {
+        private static bool IsWordSeparator(char c) => !SyntaxFacts.IsIdentifierPartCharacter(c) || c == '_';
+
+        [return: NotNullIfNotNull("name")]
+        public static string ToCleanName(string name, bool isCamelCase = true)
```

Comment:
This is moved from original `StringExtensinos.ToCleanName`.
Synced with @JoshLove-msft before, we should not expose public extension methods for BCL types, turn them into static methods intead.

Reply:
Still not convinced that we need to make this public - https://github.com/Azure/azure-sdk-for-net/pull/47730/files#r1909256494

Reply:
OK. Revert it again.